File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        nums = set(nums)
        longest = 0

        for num in nums:
          # just to make sure consective nums start from prev num of the current
          if num - 1 not in nums:
            temp = num
            while temp in nums:
              temp += 1
              longest = max(longest, temp - num)
        
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numbers = new HashSet<>();

        for (int num : nums){
          numbers.add(num);
        }
        int longest = 0;

        for (var num : numbers) {
          // Just not to miss its prev number for consecutive sequence
          if (!numbers.contains(num - 1)) {
            int y = num;
            while (numbers.contains(y)) {
              ++y;
              longest = Math.max(longest, y - num);
            }
          }
        }
        return longest;
    }
}